### PR TITLE
DOP-3988: product list for homepage

### PR DESCRIPTION
**Pull request by caesarbell**
### Stories/Links:

DOP-3988

### Current Behavior:

[Prod](https://www.mongodb.com/docs/)

### Staging Links:

[Staging](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/2992b16a3b8c3ca31df5cf9d4d9c60f20278ca6b/DOP-3988/landing/caesarbell/DOP-3988/index.html)

### Notes:

This PR ties the `..landing:products` and `..landing:product` directives into snooty, rendering the new components on the front end along with other related pre-existing components.

### CODE REVIEW

**Comment by schmalliso** [comment]
Note from design review:

Screenshot 2023-09-11 at 11 22 58 AM
There's a missing space between the copy and the arrow in the "Migrators, Tools, and Connectors" CTA.

File: src/components/Products/ProductItem.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    margin-top: 12px;
+  }
+
+  a {
+    font-size: ${FONT_SIZE};
+    font-weight: normal;
+  }
+`;
+
+const StyledH3element = styled.h3`
```
Comment (schmalliso) [inline src/components/Products/ProductItem.js:44]: Can we use the LeafyGreen typography H3 instead here?

Comment (schmalliso) [reply src/components/Products/ProductItem.js:44]: Or whatever the equivalent LG component is, pretending to be an h2

Comment (caesarbell) [reply src/components/Products/ProductItem.js:44]: @allisonmui9 yup that will be updated in this [PR](https://github.com/mongodb/snooty/pull/888)

**Comment by caesarbell** [comment]
Closing out this PR for https://github.com/mongodb/snooty/pull/888

**Pull request closed without merging**